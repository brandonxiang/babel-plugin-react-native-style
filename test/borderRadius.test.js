const pluginTester = require('babel-plugin-tester');
const reactNativeStyle = require('../src/index');

pluginTester({
  plugin: reactNativeStyle,
  pluginName: 'babel-plugin-react-native-style',
  tests: [
    {
      snapshot: true,
      title: 'border radius test',
      pluginOptions: { rpx: true, designWidth: 750 },
      code: `const styles = stylesheet.create({
          one: { borderRadius: '1 1pt 0 0' },
          two: { borderRadius: '1 1 0', },
          three: { borderRadius: '1rpx 0' }
        })`,
    },
  ],
});
