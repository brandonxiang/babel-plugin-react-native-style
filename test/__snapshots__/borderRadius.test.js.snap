// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-native-style border radius test: border radius test 1`] = `
"
const styles = stylesheet.create({
          one: { borderRadius: '1 1pt 0 0' },
          two: { borderRadius: '1 1 0', },
          three: { borderRadius: '1rpx 0' }
        })

      ↓ ↓ ↓ ↓ ↓ ↓

var __RPX = require('babel-plugin-react-native-style-runtime').__RPX;

const styles = stylesheet.create({
          one: {
                    borderTopLeftRadius: __RPX(1, 750),
                    borderTopRightRadius: 1,
                    borderBottomRightRadius: __RPX(0, 750),
                    borderBottomLeftRadius: __RPX(0, 750)
          },
          two: {
                    borderTopLeftRadius: __RPX(1, 750),
                    borderTopRightRadius: __RPX(1, 750),
                    borderBottomRightRadius: __RPX(0, 750),
                    borderBottomLeftRadius: __RPX(1, 750)
          },
          three: {
                    borderTopLeftRadius: __RPX(1, 750),
                    borderTopRightRadius: __RPX(0, 750),
                    borderBottomRightRadius: __RPX(1, 750),
                    borderBottomLeftRadius: __RPX(0, 750)
          }
});
"
`;
